packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -336,6 +340,15 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
   const defaultOnRenderVideoTile = useCallback(
     (participant: VideoGalleryRemoteParticipant, isVideoParticipant?: boolean) => {
       const remoteVideoStream = participant.videoStream;
+      // Async call to retrieve custom data model information to replace display name in remote tiles with camera on.
+      if (onFetchAvatarPersonaData) {
+        (async () => {
```

Comment:
this function is resonably complex and is nested in a complex structure (a tileRender inside a video gallery). Pull this out as a helper function
```suggestion
        fetchAvatarPersonaDataAsync()
```